File: 082_Remove_Duplicates_from_Sorted_List_II/082_2.java
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) return head;
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode curr = head;
        ListNode prev = dummy;
        while (curr != null && curr.next != null)
        {
            while (curr.next != null && curr.val == curr.next.val)
                curr = curr.next;
            if (prev.next == curr)
                prev = prev.next;
            else
                prev.next = curr.next;
            curr = curr.next;
        }
        return dummy.next;
    }    
}

File: 082_Remove_Duplicates_from_Sorted_List_II/082_2.py
class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None:
            return head

        curr = head
        prev = None
        while curr and curr.next:
            if curr.val == curr.next.val:
                currVal = curr.val
                while curr and curr.val == currVal:
                    curr = curr.next
                if prev:
                    prev.next = curr
                else:
                    head = curr
            else:
                prev = curr
                curr = curr.next
        return head